joomla/joomla-cms
# Routing: Standard-behavior for components

This PR implements a generic rule for standard behavior of components which routers extend from JComponentRouterAdvanced.
### How to test
- Apply this change
- Change the router of com_content by replacing the code with the following:
  
  ``` php
  <?php
  class ContentRouter extends JComponentRouterAdvanced
  {
      public function __construct($app = null, $menu = null) {
  
          $this->registerView('categories', 'categories');
          $this->registerView('category', 'category', 'id', 'categories', '', true, array('default', 'blog'));
          $this->registerView('article', 'article', 'id', 'category', 'catid');
          $this->registerView('archive', 'archive');
          $this->registerView('featured', 'featured');
          $this->registerView('form', 'form');
          parent::__construct($app, $menu);
  
          $this->attachRule(new JComponentRouterRulesMenu($this));
          $this->attachRule(new JComponentRouterRulesNomenu($this));
          $this->attachRule(new JComponentRouterRulesStandard($this));
      }
  
      public function getArticleKey($segments, $vars)
      {
          return (int) $segments;
      }
  }
  ```
- Comment line 69 and 123 from /components/com_content/helpers/route.php to disable the lookup of the Itemid there.
- See that it creates sane URLs and that those URLs point to the right place.
### Issues

This requires that #5446 and all depending requests are merged first. (You can still test this PR, since all changes from those PRs are included into this one.)

This was made possible through the generous donation of the people mentioned in the following link via an Indiegogo campaign: http://joomlager.de/crowdfunding/5-contributors
